src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -727,20 +728,31 @@ public virtual unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf8
                         Debug.Assert(startingAddress >= ptr && startingAddress <= (ptr + utf8Text.Length - 16));
 
                         // Load the next 16 bytes.
-                        Vector128<sbyte> sourceValue = Sse2.LoadVector128(startingAddress);
+                        Vector128<sbyte> sourceValue;
+                        bool allBytesAreAscii;
 
                         // Check for ASCII text. Any byte that's not in the ASCII range will already be negative when
                         // casted to signed byte.
-                        int index = Sse2.MoveMask(sourceValue);
+                        if (Sse2.IsSupported)
+                        {
+                            sourceValue = Sse2.LoadVector128(startingAddress);
+                            allBytesAreAscii = Sse2.MoveMask(sourceValue) == 0;
+                        }
+                        else
+                        {
+                            Debug.Assert(AdvSimd.Arm64.IsSupported);
+                            sourceValue = AdvSimd.LoadVector128(startingAddress);
+                            allBytesAreAscii = AdvSimd.Arm64.MinAcross(sourceValue).ToScalar() >= 0;
+                        }
 
-                        if (index == 0)
+                        if (allBytesAreAscii)
                         {
                             // All of the following 16 bytes is ASCII.
 
                             if (Ssse3.IsSupported)
                             {
                                 Vector128<sbyte> mask = Ssse3Helper.CreateEscapingMask(sourceValue, bitMaskLookupAsciiNeedsEscaping, bitPosLookup, nibbleMaskSByte, nullMaskSByte);
-                                index = Sse2.MoveMask(mask);
+                                int index = Sse2.MoveMask(mask);
```

Comment:
Can these two lines be optimized for `AdvSimd`?

Reply:
This particular optimization depends on the SSSE3 shuffle instruction for implementing bitmask lookups. AFAICT there's no equivalent way of achieving this in AdvSimd. @tannergooding might know if I missed something here.

Reply:
`Ssse3.Shuffle` is over bytes, which should be achievable using `AdvSimd.VectorTableLookup` and/or `AdvSimd.VectorTableLookupExtension`

Reply:
I feel that at this point there probably won't be time to work on this particular optimization. My focus is on benchmarking what we have, so I'd be inclined to just add a TODO comment and address later.